iocore/eventsystem/I_Continuation.h

```diff
@@ -71,7 +71,7 @@ constexpr ContinuationHandler
 continuation_handler_void_ptr(int (C::*fp)(int, T *))
 {
   auto fp2 = reinterpret_cast<int (C::*)(int, void *)>(fp);
-  return static_cast<ContinuationHandler>(fp2);
+  return reinterpret_cast<ContinuationHandler>(fp2);
```

Comment:
@ywkaras : Let me know whether you have thoughts about what I'm doing here. This patch is modifying code you updated to get better compile time checks (#6996). The compiler warning I'm addressing is the rather unobvious:

```
     In file included from src/ts_asf_master_test_build/iocore/eventsystem/I_Action.h:29,
                      from src/ts_asf_master_test_build/iocore/eventsystem/I_EventSystem.h:32,
                      from src/ts_asf_master_test_build/iocore/eventsystem/I_VConnection.h:29,
                      from QUICStreamVCAdapter.cc:24:
     src/ts_asf_master_test_build/iocore/eventsystem/I_Continuation.h: In instantiation of _constexpr int (Continuation::* continuation_handler_void_ptr(int (C::*)(int, T*)))(int, void*) [with C = QUICStreamVCAdapter; T = void; ContinuationHandler = int (Continuation:
     :*)(int, void*)]_:
     QUICStreamVCAdapter.cc:29:3:   required from here
     src/ts_asf_master_test_build/iocore/eventsystem/I_Continuation.h:74:10: error: left shift of negative value [-Werror=shift-negative-value]
        74 |   return static_cast<ContinuationHandler>(fp2);
           |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Changing to a reinterpret_cast from static_cast makes it happy, but if you have other ideas about how to address this, let me know.

Reply:
If we change static_cast to reinterpret_cast, there will no longer be a compile-time check that C is derived from Continuation.  If C isn't we will just see undefined behavior at runtime.  This seem clearly to be a bizarre bug in gcc on Rocky Linux, as there is no shift operator involved.  Can we change it only on Rocky Linux using conditional compilation?

Reply:
If someone could provide the contents of /etc/system-release on Rocky Linux, we could probably write an auto config macro to add '#define TS_ON_ROCKY_LINUX 0|1' in ink_config.h.

Reply:
I happened to find this conversation (not watching changes to QUIC/H3 files 😎 ). I'm totally fine with switching the order of the parents, I wonder what would be the runtime difference though.

Reply:
> I wonder what would be the runtime difference though.
It would be none, unless the compiler optimizer is able to detect cases at compile time where the address offset in the member function pointer is zero, and omit code to add it.  But, in that case, the change makes it faster, because it's causing the offset to be zero 😎 .  (Unless the the other base class besides VConnection has virtual functions, then it might be slower, but not by much.)